jina/executors/indexers/cache.py

```diff
@@ -122,11 +122,13 @@ def delete(self, keys: Iterator['UniqueId'], *args, **kwargs):
 
     def get_add_handler(self):
         # not needed, as we use the queryhandler
-        pass
+        # FIXME better way to silence warnings
+        return 1
 
     def get_query_handler(self) -> CacheHandler:
-        return self.CacheHandler(self.index_abspath, self.logger)
+        return self.CacheHandler(self.save_abspath, self.logger)
```

Comment:
are we sure about that? index_abspath makes more sense since it should contain the actual indexed data?

Reply:
Nope, that provides it a weird path, smth like `/tmp/whatever`. Using `save_abspath` provides the path to the workspace + shard dir + `name`